### Tips
+ 染色质环（contact loops）是层级最低的染色质结构，需要很高的实验质量和数据量以得到较高的分辨率，一般需要10kb以下。Loops类似于ChIP-seq的peaks，定义为Hi-C图谱中互作频率比周围相邻区域都高的格子区域。
+ 同样鉴定loops的算法也有很多并且不断更新中，较常用的比如Fit-Hi-C，Peakachu。

### Fit-Hi-C
#### Step1 hicpro2fithic
~~~
hicpro2fithic=/usr/local/software/HiC-Pro-3.1.0/bin/utils/hicpro2fithic.py
res=10000
HiCPro_anno=/home/share/HiC-Pro-annotations/

python $hicpro2fithic -i HicPro_Output/hic_results/matrix/SAMPLE/iced/${res}/SAMPLE_${res}_iced.matrix -b $HiCPro_anno/${res}_mm10.bed 
-s HicPro_Output/hic_results/matrix/SAMPLE/iced/${res}/SAMPLE_${res}_iced.matrix.biases 
-r ${res} -o 01.hicpro2fithic > 01.hicpro2fithic/SAMPLE_${res}.fithic.log

#去除线粒体
cd 01.hicpro2fithic
##拆分染色体
zcat fithic.biases.gz | awk '{print > "biases."$1}'
zcat fithic.fragmentMappability.gz | awk '{print > "fragmentMappability."$1}'
zcat fithic.interactionCounts.gz | awk '$1==$3{print > "interactionCounts."$1}'
##合并除chrM之外的染色体
mkdir ChrM
mv *chrM* ChrM/
cat biases.chr* > biases.allchr
cat fragmentMappability.chr* > fragmentMappability.allchr
cat interactionCounts.chr* > interactionCounts.allchr
##gzip
for i in *chr*; do echo $i; gzip -c $i > $i.gz & done
~~~
#### Step2 fithic_res
~~~
fithic=/home2/xycui/HIC/software/fithic-master-2.0.7/fithic-master/fithic/fithic.py

for i in {1..19} X Y;do chr=chr${i};
python $fithic -i 01.hicpro2fithic/interactionCounts.${chr}.gz 
-f 01.hicpro2fithic/fragmentMappability.${chr}.gz 
-t 01.hicpro2fithic/biases.${chr}.gz 
-l SAMPLE_${res}_${chr} -o 02.fithic_res
-r ${res} -v -p 3 -x intraOnly > 02.fithic_res/SAMPLE_${res}_${chr}_fithic.log 
done
~~~
#### Step3 significant_bin
~~~
#qvalue 0.01
for i in {1..19} X Y;do chr=chr${i}; 
zcat 02.fithic_res/SAMPLE_${res}/SAMPLE_${res}_${chr}.spline_pass2.res${res}.significances.txt.gz|
awk '$7<0.01 {print $1,$2-5000,$2+5000,$3,$4-5000,$4+5000,$5,$10,$7}'
> 03.significant_bin/SAMPLE_${res}/qvalueCutOff/SAMPLE_${res}_${chr}_qvalueCutOff.bedpe & 
done
cat 03.significant_bin/SAMPLE_${res}/qvalueCutOff/*bedpe > 03.significant_bin/SAMPLE_${res}/SAMPLE_${res}_qvalueCutOff.bedpe
~~~

### Peakachu
[Peakachu](https://github.com/tariks/peakachu)具体各参数的含义请参照示例文件和manual，该算法运算速度快。
#### Step0 Instal Peakachu
~~~
conda config --add channels defaults
conda config --add channels bioconda
conda config --add channels conda-forge
mamba create -n peakachu cooler numba scikit-learn=1.1.2 joblib=1.1.0
mamba activate peakachu
pip install -U peakachu
~~~
#### Step1 hicpro2cool
~~~
#hicConvertFormat属于hicexplorer套件，需自行安装
mamba create -n hicexplorer python=3.7
mamba install hicexplorer -c bioconda -c conda-forge
conda activate hicexplorer

hicConvertFormat -m HicPro_Output/hic_results/matrix/SAMPLLE/raw/${res}/SAMPLLE_${res}.matrix 
--inputFormat hicpro 
--bedFileHicpro $HiCPro_anno/${res}_mm10.bed 
--outputFormat cool -o 01.hicpro2cool/SAMPLE_${res}.cool 
--resolutions ${res}

#cool balance
cooler balance 01.hicpro2cool/SAMPLE_${res}.cool
~~~
#### Step2 choose_model
~~~
conda activate peakachu
peakachu depth -p 01.hicpro2cool/SAMPLE_${res}.cool > 02.choose_model/SAMPLE_${res}_sugmodel.txt
#根据SAMPLE_${res}_sugmodel.txt文件中suggested model提示信息，从Github网页下载合适的模型
#eg.suggested model: 30 million
wget http://3dgenome.fsm.northwestern.edu/peakachu/high-confidence.30million.10kb.w6.pkl -O high-confidence.30million.10kb.w6.pkl
~~~
**Pangu /home/share/HiC-Pro-annotations/peakachu/目录下已下载多个模型，可直接调用**
#### Step3 score_genome
也可单条染色体运行，命令为score_chromosome
~~~
peakachu score_genome -r ${res} --balance -p 01.hicpro2cool/SAMPLE_${res}.cool 
-O  03.score_genome/SAMPLE-peakachu-${res}-scores.bedpe 
-m $HiCPro_anno/peakachu/high-confidence.30million.10000.pkl 
~~~
#### Step4 pool
~~~
peakachu pool -r ${res} -i 03.score_genome/SAMPLE-peakachu-${res}-scores.bedpe 
-o 04.pool/SAMPLE-peakachu-${res}-loops.0.95.bedpe -t 0.95 
~~~