### Tips
+ 染色质环（contact loops）是层级最低的染色质结构，需要很高的实验质量和数据量以得到较高的分辨率，一般需要10kb以下。Loops类似于ChIP-seq的peaks，定义为Hi-C图谱中互作频率比周围相邻区域都高的格子区域。
+ 同样鉴定loops的算法也有很多并且不断更新中，较常用的比如Fit-Hi-C，Peakachu。

### Fit-Hi-C
#### Step1 hicpro2fithic
Input_matrix使用HiC-Pro结果文件中的"SAMPLE_Resolution.matrix"\
Input_bias使用HiC-Pro结果文件中的"SAMPLE_Resolution_iced.matrix.biases"\
hicpro2fithic=/usr/local/software/HiC-Pro-3.1.0/bin/utils/hicpro2fithic.py\
使用10kb分辨率：Res=10000\
HiCPro_anno=/home/share/HiC-Pro-annotations/
~~~
python $hicpro2fithic -i Input_matrix $HiCPro_anno/${Res}_mm10.bed -s $Input_bias 
-r ${Res} -o 01.hicpro2fithic/SAMPLE_${Res}> 01.hicpro2fithic/SAMPLE_${Res}/SAMPLE.fithic.log
~~~
##### 去除线粒体
~~~
cd 01.hicpro2fithic/SAMPLE_${Res}
#拆分染色体
zcat fithic.biases.gz | awk '{print > "biases."$1}'
zcat fithic.fragmentMappability.gz | awk '{print > "fragmentMappability."$1}'
zcat fithic.interactionCounts.gz | awk '$1==$3{print > "interactionCounts."$1}'
#合并除chrM之外的染色体
mkdir ChrM
mv *chrM* ChrM/
cat biases.chr* > biases.allchr
cat fragmentMappability.chr* > fragmentMappability.allchr
cat interactionCounts.chr* > interactionCounts.allchr
#gzip
for i in *chr*; do echo $i; gzip -c $i > $i.gz & done
~~~
#### Step2 fithic_res
fithic=/home2/xycui/HIC/software/fithic-master-2.0.7/fithic-master/fithic/fithic.py
~~~
for i in {1..19} X Y;do chr=chr${i};
nohup python $fithic -i 01.hicpro2fithic/SAMPLE_${Res}/interactionCounts.${chr}.gz 
-f 01.hicpro2fithic/SAMPLE_${Res}/fragmentMappability.${chr}.gz 
-t 01.hicpro2fithic/SAMPLE_${Res}/biases.${chr}.gz 
-l SAMPLE_${Res}_${chr} -o 02.fithic_res/SAMPLE_${Res} 
-r ${Res} -v -p 3 -x intraOnly > 02.fithic_res/SAMPLE_${Res}/SAMPLE_${Res}_${chr}_fithic.log 2>&1 
done
~~~
#### Step3 significant_bin
~~~
#qvalue 0.01
for i in {1..19} X Y;do chr=chr${i}; 
zcat 02.fithic_res/SAMPLE_${Res}/SAMPLE_${Res}_${chr}.spline_pass2.res${Res}.significances.txt.gz|
awk '$7<0.01 {print $1,$2-5000,$2+5000,$3,$4-5000,$4+5000,$5,$10,$7}'
> 03.significant_bin/SAMPLE_${Res}/qvalueCutOff/SAMPLE_${Res}_${chr}_qvalueCutOff.bedpe & 
done
cat 03.significant_bin/SAMPLE_${Res}/qvalueCutOff/*bedpe > 03.significant_bin/SAMPLE_${Res}/SAMPLE_${Res}_qvalueCutOff.bedpe
~~~

### Peakachu
Peakachu具体各参数的含义请参照示例文件和manual，该算法运算速度快。 \
[Peakachu Github](https://github.com/tariks/peakachu)
#### Step0 Instal Peakachu
~~~
conda config --add channels defaults
conda config --add channels bioconda
conda config --add channels conda-forge
mamba create -n peakachu cooler numba scikit-learn=1.1.2 joblib=1.1.0
mamba activate peakachu
pip install -U peakachu
~~~
#### Step1 hicpro2cool
hicConvertFormat=未安装
~~~
nohup hicConvertFormat -m Input_matrix --inputFormat hicpro --bedFileHicpro $HiCPro_anno/${Res}_mm10.bed 
--outputFormat cool -o SAMPLE_${Res}.cool --resolutions ${Res} 2>&1 &
#cool balance
nohup cooler balance SAMPLE_${Res}.cool 2>&1
~~~
#### Step2 choose_model
~~~
mamba activate peakachu
nohup peakachu depth -p SAMPLE_${Res}.cool > SAMPLE_${Res}_sugmodel.txt 2>&1 &
#根据SAMPLE_${Res}_sugmodel.txt文件中suggested model提示信息，从Github网页下载合适的模型
#eg.suggested model: 150 million
wget http://3dgenome.fsm.northwestern.edu/peakachu/high-confidence.150million.10kb.w6.pkl -O high-confidence.150million.10kb.w6.pkl
~~~
**Pangu /home/share/HiC-Pro-annotations/peakachu/目录下已下载多个模型，可直接调用**
#### Step3 score_genome
也可单条染色体运行，命令为score_chromosome
~~~
nohup peakachu score_genome -r ${Res} --balance -p SAMPLE_${Res}.cool -O SAMPLE-peakachu-${Res}-scores.bedpe 
-m high-confidence.150million.10kb.w6.pkl 2>&1 &
nohup peakachu pool -r ${Res} -i SAMPLE-peakachu-${Res}-scores.bedpe -o SAMPLE-peakachu-${Res}-loops.0.95.bedpe -t 0.95 &
~~~